portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,10 +31,22 @@
 /** Servlet that returns some example content and handles data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> commentsStorage = new ArrayList<>();
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  int max_comments_num = 3;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment");
+    PreparedQuery results = datastore.prepare(query);
+    List<String> commentsStorage = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      if (commentsStorage.size() < max_comments_num) {
```

Comment:
If the user stored more comments then expected, should we return nothing?

Reply:
What I am thinking here is: All viewers' comments are stored in datastore and "max_comments_num" is just a single viewer's preferred maximum number of comments that he/she is willing to see. So, I set it only affects the shown number, while not changing the datastore comments' contents.

Reply:
OK.